netebpfext/net_ebpf_ext_prog_info_provider.c

```diff
@@ -57,29 +51,22 @@ _net_ebpf_extension_program_info_provider_attach_client(
     _Outptr_result_maybenull_ const void** provider_dispatch)
 {
     NTSTATUS status = STATUS_SUCCESS;
-    ebpf_result_t result;
-    net_ebpf_extension_program_info_provider_t* local_provider_context =
-        (net_ebpf_extension_program_info_provider_t*)provider_context;
     net_ebpf_extension_program_info_client_t* program_info_client = NULL;
-    ebpf_extension_dispatch_table_t* client_dispatch_table;
 
     NET_EBPF_EXT_LOG_ENTRY();
 
+    UNREFERENCED_PARAMETER(provider_context);
+    UNREFERENCED_PARAMETER(client_dispatch);
     UNREFERENCED_PARAMETER(client_binding_context);
 
-    if ((provider_binding_context == NULL) || (provider_dispatch == NULL) || (local_provider_context == NULL)) {
+    if ((provider_binding_context == NULL) || (provider_dispatch == NULL)) {
         status = STATUS_INVALID_PARAMETER;
         goto Exit;
     }
 
     *provider_binding_context = NULL;
     *provider_dispatch = NULL;
 
-    client_dispatch_table = (ebpf_extension_dispatch_table_t*)client_dispatch;
```

Comment:
The PR description says this PR is a temporary workaround.  It's not clear to me if all the changes in this PR are temporary, so that it should eventually be reverted, or if some of these changes are intended to be permanent.
I might suggest creating some `#define` and wrapping all code to be reverted in ifdef's.

Reply:
_All_ changes in this PR pertain to the _new_ helper call and can be removed by reverting this PR, so I'm not sure a ```#if defined XYZ``` sandwich would buy us anything extra.  I _did_ need to manually adjust some code when I reverted PR 1827 (given code mods made since then) and maybe that's causing the apparent discrepancy? 

Or am I missing-something-in/misunderstanding your comment?